README.md

```diff
@@ -1,15 +1,225 @@
 # ros_docker_ci
 
-This repository contains the necessary scripts and docker files to set up a
-docker container that allows for CI testing in a Ubuntu/ROS environment of
-either:
+This repository encapsulates the necessary scripts and docker files to set up a
+docker container for use in Travis-CI that supports testing in a Ubuntu/ROS
+environment of either:
 
 * trusty/indigo
 * xenial/kinetic
 
-Currently, Travis-CI only natively supports up to trusty/indigo.
+An example illustrating its usage can be found at the
+[ros_docker_ci_demo](https://github.com/wavelab/ros_docker_ci_demo) repo.
 
-As many repositories depend on a CI environment that has ROS, this repository
-aims to factor out the common elements of that requirement so that separate
-docker setups do not need to be wholly maintained for every repository that uses
-them.
+## Authors
+
+* Michael Smart (Maintainer)
+* Alex Tomala
+
+## Quick Start Instructions
+
+1) Submodule this repository into the user repo:
+
+```bash
+git submodule add https://github.com/wavelab/ros_docker_ci.git REPO/RELATIVE/PATH
+```
+
+2) Create two scripts in the user repo that do CI tasks. The first is 
+`scripts/install_deps_for_docker_ci.bash` which needs to install all of the user
+repo's dependencies. The second is a script that performs the desired CI tasks
+on a container where the dependencies have been installed and the repo is
+located at root. You can specify this CI-scripts location in (3)
+
+3) Copy the `.travis.yml` file from the
+[ros_docker_ci_demo](https://github.com/wavelab/ros_docker_ci_demo) repo. Change
+all of the items in it marked as 'MODIFY_ME'.
+
+Done!
+
+## Verbose Usage Instructions
+
+An example illustrating the use of this repository can be found at
+the [ros_docker_ci_demo](https://github.com/wavelab/ros_docker_ci_demo) repo.
+
+The following instructions will replicate creating the setup used in the demo
+repo.
+
+### Submoduling
```

Comment:
Is submoduling the best way to go? Wondering if the first line or two of the repo's travis.yaml could download this repo and run a single script from inside that manages all this. That's my recent strategy with the [vtd_interface](https://github.com/wavelab/vtd_interface/blob/master/.travis.yml) and [vtd_vehicle_packages](https://github.com/wavelab/vtd_vehicle_packages/blob/master/.travis.yml) repos

Pros and cons to both, but might be nice to avoid submoduling it.